pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -112,6 +113,19 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 		return errors.Wrap(err, "failed to create csi config map")
 	}
 
+	// update the msgr2 flag
+	for _, m := range cluster.ClusterInfo.Monitors {
+		// m.Endpoint=10.1.115.104:3300
+		monPort := util.GetPortFromEndpoint(m.Endpoint)
+		if monPort == client.Msgr2port {
+			externalConnections := &cephv1.ConnectionsSpec{
+				RequireMsgr2: true,
+			}
+			cluster.Spec.Network.Connections = externalConnections
```

Comment:
This will overwrite settings from the connections if specified in the cluster CR. How about this?
```go
		if monPort == client.Msgr2port {
			if cluster.Spec.Network.Connections == nil {
				cluster.Spec.NetworkConnections = &cephv1.ConnectionsSpec{}
			}
			cluster.Spec.NetworkConnections.RequireMsgr2 = true
			break
		}
```

Reply:
Okay, 

But using a `externalConnections` variable I think we should do, to avoid a nil pointer possiblity

Reply:
I realized, the `cluster CR` should not be given the priority for this case,
As we already have a mon using the port v2 and other two mons using the v1 port, 

So to make that consistent either we should make all three mons to v2 or v1, and for the above scenario I guess we should force move it to v2

Reply:
We really should expect that all mons are using the same port, otherwise it would be confusing whether they are using msgr1 or msgr2. I don't think we should fail the reconcile if the ports are different, so maybe just log a warning in that case. And then we should only set requireMsgr2 if all ports are on the v2 port.

Reply:
> But using a externalConnections variable I think we should do, to avoid a nil pointer possiblity

What nil pointer is possible with that approach? I had a typo in my code suggestion above, fixed now.  The problem currently with your implementation is that any settings in the cluster CR under the network connections would be lost when you create the connectionsSpec. Perhaps those settings wouldn't apply to the external scenario, but it's good practice to make sure we don't lose those settings from the CR.

Reply:
>The problem currently with your implementation is that any settings in the cluster CR under the network connections would be lost when you create the connectionsSpec

Got your point, thanks for the feedback